tensorflow_addons/layers/optical_flow_test.py

```diff
@@ -74,51 +74,50 @@ def _create_test_data(data_format):
     return val_a, val_b
 
 
-@test_utils.run_all_in_graph_and_eager_modes
-class CorrelationCostTest(tf.test.TestCase):
-    def _forward_simple(self, data_format, use_gpu=False):
-        # We are just testing where the output has vanishing values.
-        with test_utils.device(use_gpu):
-            val_a, val_b = _create_test_data(data_format)
-            input_a = tf.constant(val_a, dtype=tf.float32)
-            input_b = tf.constant(val_b, dtype=tf.float32)
-
-            input_a_tensor = tf.convert_to_tensor(input_a, dtype=tf.float32)
-            input_b_tensor = tf.convert_to_tensor(input_b, dtype=tf.float32)
-
-            kernel_size = 1
-            max_displacement = 2
-            stride_1 = 1
-            stride_2 = 2
-            pad = 4
+@pytest.mark.usefixtures("cpu_and_gpu", "maybe_run_functions_eagerly")
+def test_forward_simple(data_format):
```

Comment:
I can't say I find it very intuitive that this is a shared fixture being loaded from conftest. If the documentation will be expanded to describe the pytest method I'm okay with it though.

Reply:
Yes definitly. Pytest has many shortcuts that are impossible to understand unless you read the docs. I can't say I agree with all the design choices, but to me there is no doubt that it's much easier to use than unittest.

We should have a section in the CONTRIBUTING.md explaining this machinery. I agree, it's work in progress.